### PR TITLE
Wyvern exchange subgraph

### PR DESCRIPTION
**Pull request by anshulforyou**

### CODE REVIEW

File: seaport-subgraph/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	type: AirTransactionType!
+  index: BigInt!
+  protocolType: AirProtocolType!
+  protocolActionType: AirProtocolActionType!
+	tokenId: BigInt! #nft
+  tokenAmount:BigInt!   #nft 
+	transactionToken: AirToken!   #nft
+	paymentToken: AirToken #payment
+  paymentAmount: BigInt #payment
+  royaltyAmount: BigInt
```
Comment (0xsarvesh) [inline seaport-subgraph/schema.graphql:139]: Please upgrade to new schema

File: wyvern-exchange/schema.graphql [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	type: AirTransactionType!
+  index: BigInt!
+  protocolType: AirProtocolType!
+  protocolActionType: AirProtocolActionType!
+	tokenId: BigInt! #nft
+  tokenAmount:BigInt!   #nft 
+	transactionToken: AirToken!   #nft
+	paymentToken: AirToken #payment
+  paymentAmount: BigInt #payment
+  royaltyAmount: BigInt
```
Comment (0xsarvesh) [inline wyvern-exchange/schema.graphql:139]: Can you upgrade to new schema.

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]

File: wyvern-exchange/src/wyvern-exchange.ts [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+      sellOrder.callData!,
+      buyOrder.replacementPattern!
+    );
+
+    for (let i = 0; i < decoded.transfers.length; i++) {
+      log.info("txHash {} transfer method {}", [txHash.toHexString(), decoded.transfers[i].method]);
+      let nft = new airstack.nft.NFT(decoded.addressList[i], decoded.transfers[i].method, decoded.transfers[i].token, decoded.transfers[i].amount);
+      let feeAmount = BigInt.zero();
+      let feeRecipient = "";
+      let creatorRoyaltyFeePercentage = BigInt.zero();
```
Comment (0xsarvesh) [inline wyvern-exchange/src/wyvern-exchange.ts:136]: Remove unused variables.

File: wyvern-exchange/src/wyvern-exchange.ts [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+      let totalRevenueETH = BigInt.zero();
+      let marketplaceRevenueETH = BigInt.zero();
+      let creatorRevenueETH = BigInt.zero();
```
Comment (0xsarvesh) [inline wyvern-exchange/src/wyvern-exchange.ts:139]: Remove unused variables

File: wyvern-exchange/src/wyvern-exchange.ts [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+      if (sellOrder.feeRecipient!=Address.zero().toHexString()) {
+        creatorRoyaltyFeePercentage = EXCHANGE_MARKETPLACE_FEE.le(
+          sellOrder.makerRelayerFee
+        )
+          ? sellOrder.makerRelayerFee
```
Comment (0xsarvesh) [inline wyvern-exchange/src/wyvern-exchange.ts:144]: Please extract the functions for common logic. 
If & else has same logic except variable change, it code duplication.

File: wyvern-exchange/src/wyvern-exchange.ts [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+        creatorRevenueETH = totalRevenueETH.minus(marketplaceRevenueETH);
+  
+        feeRecipient =sellOrder.feeRecipient;
+      }else{
+      // if (buyOrder.feeRecipient != Address.zero().toHexString()){
+        creatorRoyaltyFeePercentage = EXCHANGE_MARKETPLACE_FEE.le(
+          buyOrder.takerRelayerFee
+        )
+          ? buyOrder.takerRelayerFee
+              .minus(EXCHANGE_MARKETPLACE_FEE)
```
Comment (0xsarvesh) [inline wyvern-exchange/src/wyvern-exchange.ts:168]: Again same code repetition.

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]

File: wyvern-exchange/src/wyvern-exchange.ts (hunk 3 above)
Comment (anshulforyou) [reply wyvern-exchange/src/wyvern-exchange.ts:136]: It is being used in the code.

File: wyvern-exchange/src/wyvern-exchange.ts (hunk 4 above)
Comment (anshulforyou) [reply wyvern-exchange/src/wyvern-exchange.ts:139]: This one as well

File: seaport-subgraph/schema.graphql [hunk 7]
```diff
@@ -0,0 +1,164 @@
+enum AirNetwork {
```
[the remark this answers is not in the document]
Comment (anshulforyou) [reply seaport-subgraph/schema.graphql:1]: Empty folder exists, would be resolved in merge

File: wyvern-exchange/src/wyvern-exchange.ts (hunk 3 above)
Comment (anshulforyou) [reply wyvern-exchange/src/wyvern-exchange.ts]: Refactored the code, only necessary variables are being used now

File: wyvern-exchange/src/wyvern-exchange.ts (hunk 4 above)
Comment (anshulforyou) [reply wyvern-exchange/src/wyvern-exchange.ts]: Refactored the code, only necessary variables are being used now

File: wyvern-exchange/src/wyvern-exchange.ts (hunk 5 above)
Comment (anshulforyou) [reply wyvern-exchange/src/wyvern-exchange.ts]: Done, created a function for it

**Review by 0xsarvesh — APPROVED** [review]

**Pull request merged**